relay/relay.go

```diff
@@ -43,22 +43,24 @@ func UnmarshalSpec(id graphql.ID, v interface{}) error {
 	return json.Unmarshal([]byte(s[i+1:]), v)
 }
 
+type RelayInput struct {
+	Query         string                 `json:"query"`
+	OperationName string                 `json:"operationName"`
+	Variables     map[string]interface{} `json:"variables"`
+}
+
 type Handler struct {
 	Schema *graphql.Schema
+	Input  RelayInput
```

Comment:
This is not a good idea as `Input` gets reused between requests. As a matter of fact, this will 100% lead to concurrency bugs.